java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -122,6 +130,38 @@ class DatabaseImpl(
         }
     }
 
+    /**
+     * Returns the ID for the given [StorageKey] if one already exists, otherwise creates a new one
+     * for it.
+     */
+    @VisibleForTesting
+    fun getStorageKeyId(storageKey: StorageKey): StorageKeyId {
```

Comment:
It could be worth adding a TODO to consider using an LRU cache of storageKeys->StorageKeyIds to speed up the most-frequently-updated operations.

Reply:
Done